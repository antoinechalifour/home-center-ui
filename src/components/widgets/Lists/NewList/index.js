import { graphql } from 'react-apollo'
import { createList } from 'queries/lists'
import NewList from './NewList'

const queryOptions = {}

export default graphql(createList, queryOptions)(NewList)
