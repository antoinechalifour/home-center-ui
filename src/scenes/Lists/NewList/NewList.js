import React, { Component } from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'

const Input = styled.input`
  box-sizing: border-box;
  display: block;
  width: 100%;
  margin: 16px auto;
  background: rgba(0, 0, 0, .8);
  border: none;
  font-size: inherit;
  font-family: inherit;
  padding: 8px 12px;
  border-radius: 4px;
  color: #fff;
  outline: none;
`

export default class NewList extends Component {
  static propTypes = {
    mutate: PropTypes.func.isRequired
  }

  state = {
    value: ''
  }

  onChange = e => this.setState({ value: e.target.value })

  onSubmit = e => {
    e.preventDefault()

    this.props.mutate({ variables: { name: this.state.value } })
    this.setState({ value: '' })
  }

  render () {
    return (
      <form onSubmit={this.onSubmit}>
        <Input
          value={this.state.value}
          onChange={this.onChange}
          placeholder='New list?'
        />
      </form>
    )
  }
}
