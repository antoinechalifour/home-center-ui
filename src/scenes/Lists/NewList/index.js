import { graphql } from 'react-apollo'
import { createList } from 'queries/lists'
import NewList from './NewList'

const queryOptions = {
  options: {
    refetchQueries: ['GetLists']
  }
}

export default graphql(createList, queryOptions)(NewList)
