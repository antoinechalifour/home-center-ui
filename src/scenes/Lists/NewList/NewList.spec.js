import 'jest-styled-components'
import React from 'react'
import renderer from 'react-test-renderer'
import { shallow } from 'enzyme'
import NewList from './NewList'

describe('scenes/Lists/NewList', () => {
  it('Should render correctly', () => {
    const props = {
      mutate: jest.fn()
    }
    const tree = renderer.create(<NewList {...props} />)

    tree.getInstance().setState({ value: 'my new list!' })

    expect(tree).toMatchSnapshot()
  })

  it('Should call the mutation when the form is submitted', () => {
    const props = {
      mutate: jest.fn()
    }
    const wrapper = shallow(<NewList {...props} />)
    const preventDefault = jest.fn()

    wrapper.setState({ value: 'my #2 item' })
    wrapper.simulate('submit', { preventDefault })

    expect(props.mutate.mock.calls.length).toBe(1)
    expect(props.mutate.mock.calls[0][0]).toEqual({
      variables: {
        input: { name: 'my #2 item' }
      }
    })
    expect(wrapper.state()).toEqual({ value: '' })
  })
})
