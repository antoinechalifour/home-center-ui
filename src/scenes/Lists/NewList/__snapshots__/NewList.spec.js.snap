// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`scenes/Lists/NewList Should render correctly 1`] = `
<styled.form
  onSubmit={[Function]}
>
  <styled.input
    onChange={[Function]}
    placeholder="Add a new list"
    value="my new list!"
  />
</styled.form>
`;
